watertap/core/zero_order_costing.py

```diff
@@ -1518,62 +1517,71 @@ def cost_nanofiltration(blk):
-            mem_cost, rep_rate = _get_tech_parameters(
-                blk,
-                parameter_dict,
-                blk.unit_model.config.process_subtype,
-                ["membrane_cost", "membrane_replacement_rate"])
-
-            # Determine if a costing factor is required
-            factor = parameter_dict["capital_cost"]["cost_factor"]
-
-            # Add cost variable and constraint
-            blk.capital_cost = pyo.Var(
-                initialize=1,
-                units=blk.config.flowsheet_costing_block.base_currency,
-                bounds=(0, None),
-                doc="Capital cost of unit operation")
-
-            blk.fixed_operating_cost = pyo.Var(
-                initialize=1,
-                units=blk.config.flowsheet_costing_block.base_currency,
-                bounds=(0, None),
-                doc="Fixed operating cost of unit operation")
-
-            capex_expr = pyo.units.convert(
-                mem_cost * pyo.units.convert(blk.unit_model.area, to_units=pyo.units.m ** 2),
-                to_units=blk.config.flowsheet_costing_block.base_currency)
-
-            if factor == "TPEC":
-                capex_expr *= blk.config.flowsheet_costing_block.TPEC
-            elif factor == "TIC":
-                capex_expr *= blk.config.flowsheet_costing_block.TIC
-
-            blk.capital_cost_constraint = \
-                pyo.Constraint(expr=blk.capital_cost == capex_expr)
-
-            blk.fixed_operating_cost_constraint = \
-                pyo.Constraint(expr=blk.fixed_operating_cost ==
-                                    pyo.units.convert(rep_rate
-                                                      * mem_cost
-                                                      * pyo.units.convert(
-                                                      blk.unit_model.area, to_units=pyo.units.m ** 2),
-                                                      to_units=blk.config.flowsheet_costing_block.base_currency))
+        blk.fixed_operating_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency,
+            bounds=(0, None),
+            doc="Fixed operating cost of unit operation")
+
+        capex_expr = pyo.units.convert(
+            mem_cost * pyo.units.convert(blk.unit_model.area, to_units=pyo.units.m ** 2),
+            to_units=blk.config.flowsheet_costing_block.base_currency)
+
+        # Determine if a costing factor is required
+        factor = parameter_dict["capital_cost"]["cost_factor"]
+        if factor == "TPEC":
+            capex_expr *= blk.config.flowsheet_costing_block.TPEC
+        elif factor == "TIC":
+            capex_expr *= blk.config.flowsheet_costing_block.TIC
+
+        blk.capital_cost_constraint = \
+            pyo.Constraint(expr=blk.capital_cost == capex_expr)
+
+        blk.fixed_operating_cost_constraint = \
+            pyo.Constraint(expr=blk.fixed_operating_cost ==
```

Comment:
A couple of comments regarding consistency with other zero-order unit operations:

1. Is the really a fixed cost, or a variable cost (i.e. strictly replacement is a function of usage).
2. In other similar models, we have generally just registered material replacement as a flow (even if it is a batch replacement process).

Reply:
I suppose that it's technically a variable cost. I just implemented it the same way that it was implemented in `watertap_costing`. 

I noticed that we registered as flows for everything else,  but thought this might be a different category then the rest (all other flows are either per unit kg or per kWh).

Reply:
I changed this to a variable cost instead.